lib/ex_aws/request/url.ex

```diff
@@ -17,6 +18,16 @@ defmodule ExAws.Request.Url do
     |> String.trim_trailing("?")
   end
 
+  defp set_host(config, operation) do
+    case operation do
+      %ExAws.Operation.S3{} ->
+        %{config | host: "#{operation.bucket}.s3.amazonaws.com"}
```

Comment:
Assuming the host suffix is `s3.amazonaws.com` is invalid. In an app I have the host suffix is `s3.nl-ams.scw.cloud`.

This should not be done here anyways, imho, but back in `operations/s3.ex` where it used to be.